lnwire/dyn_propose.go

```diff
@@ -0,0 +1,316 @@
+
+// Encode serializes the target DynPropose into the passed io.Writer.
+// Serialization will observe the rules defined by the passed protocol version.
+//
+// This is a part of the lnwire.Message interface.
+func (dp *DynPropose) Encode(w *bytes.Buffer, _ uint32) error {
+	var tlvRecords []tlv.Record
+	if dp.DustLimit != nil {
+		dustLimit := uint64(*dp.DustLimit)
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPDustLimitSatoshis, &dustLimit,
+			),
+		)
+	}
+	if dp.MaxValueInFlight != nil {
+		maxValue := uint64(*dp.MaxValueInFlight)
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPMaxHtlcValueInFlightMsat, &maxValue,
+			),
+		)
+	}
+	if dp.ChannelReserve != nil {
+		channelReserve := uint64(*dp.ChannelReserve)
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPChannelReserveSatoshis, &channelReserve,
+			),
+		)
+	}
+	if dp.CsvDelay != nil {
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPToSelfDelay, dp.CsvDelay,
+			),
+		)
+	}
+	if dp.MaxAcceptedHTLCs != nil {
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPMaxAcceptedHtlcs, dp.MaxAcceptedHTLCs,
+			),
+		)
+	}
+	if dp.FundingKey != nil {
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPFundingPubkey, &dp.FundingKey,
+			),
+		)
+	}
+	if dp.ChannelType != nil {
+		tlvRecords = append(
+			tlvRecords, tlv.MakeDynamicRecord(
+				DPChannelType, dp.ChannelType,
+				dp.ChannelType.featureBitLen,
+				channelTypeEncoder, channelTypeDecoder,
+			),
+		)
+	}
+	if dp.KickoffFeerate != nil {
+		kickoffFeerate := *dp.KickoffFeerate
+		tlvRecords = append(
+			tlvRecords, tlv.MakePrimitiveRecord(
+				DPKickoffFeerate, &kickoffFeerate,
+			),
+		)
+	}
+	tlv.SortRecords(tlvRecords)
+
+	tlvStream, err := tlv.NewStream(tlvRecords...)
+	if err != nil {
+		return err
+	}
+
+	var extraBytesWriter bytes.Buffer
+	if err := tlvStream.Encode(&extraBytesWriter); err != nil {
+		return err
+	}
+
+	dp.ExtraData = ExtraOpaqueData(extraBytesWriter.Bytes())
+
+
+	if err := WriteChannelID(w, dp.ChanID); err != nil {
+		return err
+	}
+
+	if err := WriteBool(w, dp.Initiator); err != nil {
+		return err
+	}
+
+	return WriteBytes(w, dp.ExtraData)
+}
+
+// Decode deserializes the serialized DynPropose stored in the passed io.Reader
+// into the target DynPropose using the deserialization rules defined by the
+// passed protocol version.
+//
+// This is a part of the lnwire.Message interface.
+func (dp *DynPropose) Decode(r io.Reader, _ uint32) error {
+	// Parse out the only required field.
+	if err := ReadElements(r, &dp.ChanID, &dp.Initiator); err != nil {
+		return err
+	}
+
+	// Parse out TLV stream.
+	var tlvRecords ExtraOpaqueData
+	if err := ReadElements(r, &tlvRecords); err != nil {
+		return err
+	}
+
+	// Prepare receiving buffers to be filled by TLV extraction.
+	var dustLimitScratch uint64
+	dustLimit := tlv.MakePrimitiveRecord(DPDustLimitSatoshis,
```

Comment:
For instances like this we usually prefer a clean line break, so: 
```go
	maxValue := tlv.MakePrimitiveRecord(
	        DPMaxHtlcValueInFlightMsat, &maxValueScratch,
	)
```

Applies to all instances below this.